File: mobile/lib/services/stripe_service.dart
import 'api_client.dart';

class StripeIntent {
  final String clientSecret;
  final String paymentIntentId;
  StripeIntent({required this.clientSecret, required this.paymentIntentId});

  factory StripeIntent.fromJson(Map<String, dynamic> json) => StripeIntent(
        clientSecret: json['clientSecret'],
        paymentIntentId: json['paymentIntentId'],
      );
}

/// Calls the backend to create a Stripe PaymentIntent. The clientSecret is what you'd
/// pass to the flutter_stripe SDK to collect card details and confirm on-device --
/// see README "Next steps" for wiring that up.
class StripeService {
  final ApiClient _client = ApiClient();

  Future<StripeIntent> createIntent(double amount, {String currency = 'usd'}) async {
    final data = await _client.post('/payments/stripe/create-intent', {
      'amount': amount,
      'currency': currency,
    });
    return StripeIntent.fromJson(data);
  }
}

File: web-admin/src/app/core/services/stripe.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';

export interface CreateIntentResponse {
  clientSecret: string;
  paymentIntentId: string;
}

/// <summary>
/// Calls the backend to create a Stripe PaymentIntent (see StripeIntentController).
/// The returned clientSecret is what you pass to Stripe.js/Stripe Elements on the
/// client to collect card details and confirm the payment -- see README "Next steps"
/// for wiring up the actual Stripe Elements card form.
/// </summary>
@Injectable({ providedIn: 'root' })
export class StripeService {
  private http = inject(HttpClient);

  createIntent(amount: number, currency = 'usd'): Observable<CreateIntentResponse> {
    return this.http.post<CreateIntentResponse>(`${environment.apiBaseUrl}/payments/stripe/create-intent`, { amount, currency });
  }
}
